{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "aa857aae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Après transformation:\n",
      "          ID     ACCESSION                                         DEFINITION  \\\n",
      "0  513136972  KF177113\\r\\n  Enterovirus B83 isolate A40-37PD VP1-like gene...   \n",
      "1  513136971  KF177112\\r\\n  Enterovirus B83 isolate A59-35PI VP1-like gene...   \n",
      "2  513136970  KF177111\\r\\n  Enterovirus A76 isolate A29-23PI VP1-like gene...   \n",
      "3  937628841  KR065422\\r\\n  Enterovirus B strain PAK_NIH_2009-23 capsid pr...   \n",
      "4  667795969  KF413055\\r\\n  Human enterovirus C99 isolate NIV0512266 VP1 g...   \n",
      "\n",
      "          VERSION                 REFERENCE       ORGANISM  \\\n",
      "0  KF177113.1\\r\\n   1  (bases 1 to 864)\\r\\n  Enterovirus B   \n",
      "1  KF177112.1\\r\\n   1  (bases 1 to 466)\\r\\n  Enterovirus B   \n",
      "2  KF177111.1\\r\\n   1  (bases 1 to 261)\\r\\n  Enterovirus A   \n",
      "3  KR065422.1\\r\\n  1  (bases 1 to 2574)\\r\\n  Enterovirus B   \n",
      "4  KF413055.1\\r\\n   1  (bases 1 to 320)\\r\\n  Enterovirus C   \n",
      "\n",
      "                                             AUTHORS  \\\n",
      "0  Rao,D.C., Reddy,H., Sudheendra,K., Raghavendra...   \n",
      "1  Rao,D.C., Reddy,H., Sudheendra,K., Raghavendra...   \n",
      "2  Rao,D.C., Reddy,H., Sudheendra,K., Raghavendra...   \n",
      "3  Shaukat,S., Angez,M., Alam,M.M., Sharif,S., Kh...   \n",
      "4  Patil,P.R., Chitambar,S.D. and Gopalkrishna,V....   \n",
      "\n",
      "                                               TITLE  \\\n",
      "0  Non-polio enterovirus association with persist...   \n",
      "1  Non-polio enterovirus association with persist...   \n",
      "2  Non-polio enterovirus association with persist...   \n",
      "3  Characterization of non-polio enterovirus isol...   \n",
      "4  Molecular surveillance of non-polio enteroviru...   \n",
      "\n",
      "                     Assembly.Method          Sequencing.Technology  \\\n",
      "0  GS De Novo Assembler v. v 2.6\\r\\n                        454\\r\\n   \n",
      "1  GS De Novo Assembler v. v 2.6\\r\\n                        454\\r\\n   \n",
      "2  GS De Novo Assembler v. v 2.6\\r\\n                        454\\r\\n   \n",
      "3                                NaN  Sanger dideoxy sequencing\\r\\n   \n",
      "4                                NaN  Sanger dideoxy sequencing\\r\\n   \n",
      "\n",
      "    Location.Qualifiers     mol_type           strain          host   country  \\\n",
      "0            1..864\\r\\n  genomic RNA              NaN  Homo sapiens     India   \n",
      "1            1..466\\r\\n  genomic RNA              NaN  Homo sapiens     India   \n",
      "2            1..261\\r\\n  genomic RNA              NaN  Homo sapiens     India   \n",
      "3           1..2574\\r\\n  genomic RNA  PAK_NIH_2009-23  Homo sapiens  Pakistan   \n",
      "4            1..320\\r\\n  genomic RNA              NaN  Homo sapiens     India   \n",
      "\n",
      "  collection_date  Date_Soumission  Sequence_Length standardized_date  \n",
      "0            2010  14-AUG-2017\\r\\n              864        2010-01-01  \n",
      "1   \"16-May-2010\"  14-AUG-2017\\r\\n              466        2010-05-16  \n",
      "2   \"22-Jun-2009\"  14-AUG-2017\\r\\n              261        2009-06-22  \n",
      "3            2009  14-OCT-2015\\r\\n              574        2009-01-01  \n",
      "4          \"2005\"  30-JUL-2014\\r\\n              320        2005-01-01  \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Chargement du fichier CSV\n",
    "csv_path = \"NPEV_ABCD_clean_dated.csv\"\n",
    "df = pd.read_csv(csv_path, sep=';')\n",
    "\n",
    "# Fonction pour convertir les dates au format 'YYYY/MM/DD'\n",
    "def convert_date_format(date):\n",
    "    try:\n",
    "        # Tentative de conversion des dates 'YYYY-MM-DD'\n",
    "        date_converted = pd.to_datetime(date, format='%Y-%m')\n",
    "    except ValueError:\n",
    "        try:\n",
    "            # Tentative de conversion des dates 'DD/MM/YYYY'\n",
    "            date_converted = pd.to_datetime(date, format='%d/%m/%Y')\n",
    "        except ValueError:\n",
    "            return date  # Retourne la date telle quelle si les deux conversions échouent\n",
    "    \n",
    "    return date_converted.strftime('%Y/%m/%d')\n",
    "\n",
    "# Application de la fonction de conversion à la colonne 'collection_date'\n",
    "df['standardized_date'] = df['standardized_date'].apply(convert_date_format)\n",
    "\n",
    "# Affichage du dataframe après transformation\n",
    "print(\"Après transformation:\")\n",
    "print(df.head())\n",
    "\n",
    "# Enregistrement du dataframe transformé dans un nouveau fichier CSV\n",
    "output_path = \"NPEV_ABCD_clean_datedYYYMMDD.csv\"\n",
    "df.to_csv(output_path, sep=';', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3980f0f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the cleaned DataFrame to a new CSV file\n",
    "df.to_csv('NPEV_ABCD_clean_dated.csv', index=False, sep=';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cf00463",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
